event.go

```diff
@@ -0,0 +1,376 @@
+	EventData
+	Cause     int    `json:"cause"`
+	Cause_txt string `json:"cause_txt"`
+	ChannelData
+}
+
+// ChannelDialplan events indicate a channel has changed its location in the dialplan
+//   NOTE: This event also most likely implies the channel is leaving the control of this
+//   ARI application
+type ChannelDialplan struct {
+	EventData
+	ChannelData
+	Dialplan_app      string `json:"dialplan_app"`      // The application which is to be executed
+	Dialplan_app_data string `json:"dialplan_app_data"` // The data to be passed to the application
+}
+
+// ChannelDtmfReceived events indicate a channel has received a DTMF tone.
+//   NOTE: this event is sent at the _end_ of the DTMF tone (and there is no indication
+//   for the _start_ of the DTMF tone)
+type ChannelDtmfReceived struct {
+	EventData
+	ChannelData
+	Digit       string `json:"digit"`       // The DTMF digit which was received (0-9A-E*#)
+	Duration_ms int    `json:"duration_ms"` // The duration of the DTMF tone, in milliseconds
+}
+
+// ChannelEngteredBridge events indicate that a channel has joined a bridge
+type ChannelEnteredBridge struct {
+	EventData
+	Bridge Bridge `json:"bridge"`
+	ChannelData
+}
+
+// ChannelHangupRequest events indicate that a channel has received a hangup
+// request
+//   TODO: find out if the application is supposed to act on this, or whether
+//   this event is purely advisory
+type ChannelHangupRequest struct {
+	EventData
+	Cause int `json:"cause,omitempty"` // Integer cause code
+	ChannelData
+	Soft bool `json:"soft,omitempty"` // Whether the request was "soft"
+}
+
+// ChannelHold events indicate that a channel has been put on hold
+type ChannelHold struct {
+	EventData
+	ChannelData
+	MusicClass string `json:"musicclass"` // The music class being played to the held channel
+}
+
+// ChannelLeftBridge events indicate that a channel has left a bridge
+type ChannelLeftBridge struct {
+	EventData
+	Bridge Bridge `json:"bridge"`
+	ChannelData
+}
+
+// ChannelStateChange events indicate that the state of a channel has changed
+//   TODO: enumerate the possible channel states
+type ChannelStateChange struct {
+	EventData
+	ChannelData
+}
+
+// ChannelTalkingFinished events indicate that previously-detected talking on
+// a channel is now absent
+type ChannelTalkingFinished struct {
+	EventData
+	ChannelData
+	Duration int `json:"duration"` // Duration (in milliseconds) of the talking
+}
+
+// ChannelTalkingStarted events indicate that talking has been detected on a channel
+type ChannelTalkingStarted struct {
+	EventData
+	ChannelData
+}
+
+// ChannelUnhold events indicate that a channel has removed from Hold
+type ChannelUnhold struct {
+	EventData
+	ChannelData
+}
+
+// ChannelUserevent events are custom-formatted events which have been received
+//   TODO: figure out if these include AMI user events, etc, and how the data is
+//   formatted
+type ChannelUserevent struct {
+	EventData
+	Bridge Bridge `json:"bridge,omitempty"`
```

Comment:
Done in https://github.com/CyCoreSystems/ari/pull/28/commits/6067360c2151c6dfad49c214e58a3bd54b95977d